src/lib/ui/core/Dropdown/Dropdown.svelte

```diff
@@ -0,0 +1,99 @@
+<script lang="ts" generics="T extends {icon?: string} | string">
+  import type { ComponentProps, Snippet } from 'svelte'
+
+  import Button from '$ui/core/Button/Button.svelte'
+  import Popover from '$ui/core/Popover/Popover.svelte'
+  import { cn } from '$ui/utils/index.js'
+
+  type TProps = {
+    class?: string
+    items: T[]
+    valueKey?: keyof T
+    selected?: T
+    label?: Snippet<[T]>
+    trigger?: Snippet
+    triggerClass?: string
+    onSelect: (item: T) => void
+    closeDelay?: number
+  } & Omit<ComponentProps<typeof Popover>, 'content' | 'children'>
+
+  let {
+    class: className,
+    items,
+    valueKey,
+    selected = $bindable(),
+    trigger,
+    triggerClass,
+    label = defaultLabel,
+    onSelect,
+    closeDelay = 0,
+    ...rest
+  }: TProps = $props()
+
+  let isOpened = $state(false)
+  let closeTimer = $state<NodeJS.Timeout>()
+
+  function onItemSelect(item: T, close: () => void) {
+    clearTimeout(closeTimer)
+    selected = item
+    onSelect(item)
+
+    closeTimer = setTimeout(() => {
+      close()
+    }, closeDelay)
+  }
```

Comment:
`closeTimer` is not used inside markup or in any effects, so it's safe to assign it to a local non-state variable
```suggestion
  let closeTimer: NodeJS.Timeout

  function onItemSelect(item: T, close: () => void) {
    clearTimeout(closeTimer)
    selected = item
    onSelect(item)

    closeTimer = setTimeout(close, closeDelay)
  }
```